mobile/src/main/java/org/openhab/habdroid/ui/widget/WidgetImageView.java

```diff
@@ -308,12 +304,15 @@ public void execute(boolean avoidCache) {
         public void cancel() {
             if (mCall != null) {
                 mCall.cancel();
-                mCall = null;
             }
         }
 
+        public boolean hasCompleted() {
+            return mCall == null;
+        }
+
         public boolean isActiveForUrl(HttpUrl url) {
-            return mCall != null && mCall.request().url().equals(url);
+            return mCall != null && mCall.request().url().equals(url) && !mCall.isCanceled();
```

Comment:
`return !hasCompleted() ...`?

Reply:
I went a bit back and forth about this (started splitting the call to `isActiveForUrl()` into `!hasCompleted() && matchesUrl(url)`), but ultimately figured keeping it the way it is is the most readable/simple solution. If you have strong feelings about it though, I can change it.